### PR TITLE
163 since format is printed, we may want to improve how formats are printed, currently it is

### LINKED ISSUE #163: since format is printed, we may want to improve how formats are printed, currently it is
Since format is printed, we may want to improve how formats are printed, currently it is 

```
Mapping of:
A  <-  a 
NN  <-  NA 
NA <-   
Convert to factor: TRUE 
Drop unused level: FALSE 
NA-replacing level in last position: TRUE 
```

right hand side of NA is empty, is it empty string?
and we can not distinguish between "NA" string and `NA_character`. maybe just follow the `forcats` to use `<NA>` to represent the `NA_character`?

_Originally posted by @clarkliming in https://github.com/insightsengineering/dunlin/pull/162#discussion_r1794866823_

### PR DESCRIPTION
**Pull request by BFalquet**
https://github.com/insightsengineering/dunlin/issues/163

close #163 

Improve the print method to display `<NA>` when the value is `NA` and surround strings with `"` to highlight the fact that they are character and allow to better visualize with spaces.

thank you for the review

### CODE REVIEW

**Comment by BFalquet** [comment]
@clarkliming I updated the enclosing quotes with `stringr` if you can have a look. Thank you in advance.

File: tests/testthat/_snaps/rules.md [hunk 1]
```diff
@@ -4,9 +4,9 @@
       rule(a = "1", b = NA)
     Output
       Mapping of:
-      a  <-  1 
-      b  <-  NA 
-      NA <-   
+      a  <-  "1" 
+      b  <-  <NA> 
+      Convert to <NA>: ""
```
Comment (clarkliming) [inline tests/testthat/_snaps/rules.md:9]: can we have one to many matching rule?

File: R/rules.R [hunk 2]
```diff
@@ -52,12 +52,16 @@ print.rule <- function(x, ...) {
   if (length(x) == 0) {
     cat("Empty mapping.\n")
   } else {
-    for (i in seq_len(length(x))) {
-      cat(nms[i], " <- ", if (length(x[[i]]) > 1) sprintf("[%s]", toString(x[[i]])) else x[[i]], "\n")
+    for (i in seq_along(x)) {
+      ori_nms <- if (length(x[[i]]) > 1) sprintf("[%s]", toString(x[[i]])) else x[[i]]
```
Comment (clarkliming) [inline R/rules.R:56]: is this line makeing a difference? (in the snapshots there seems no `[]` )

Comment (BFalquet) [reply R/rules.R:56]: that seems to be redundant, plus now we only accept characters so I don't think the length can be >1. This is probably a residual from previous design.

Comment (BFalquet) [reply R/rules.R:56]: I think that would make sense if we were printing the rules like that:
```
x <- "a", "b"
```
but currently we are doing:
```
x <- "a"
x <- "b"
```

Should we change ?

Comment (clarkliming) [reply R/rules.R:56]: I think we can update it, this is a bit clearer

**Review by clarkliming — APPROVED** [review]
looks good to me!

**Pull request merged**